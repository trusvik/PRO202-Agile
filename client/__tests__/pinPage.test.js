import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom'; // to handle routing context
import PinPage from '../components/user/pinPage';

// Mock the `useNavigate` hook
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => jest.fn(),
}));

describe('PinPage Snapshot Test', () => {
    it('renders correctly', () => {
        const setIsAuthenticated = jest.fn(); // Mock function for setIsAuthenticated

        // Render the component
        const tree = renderer.create(
            <MemoryRouter>
                <PinPage setIsAuthenticated={setIsAuthenticated} />
            </MemoryRouter>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
