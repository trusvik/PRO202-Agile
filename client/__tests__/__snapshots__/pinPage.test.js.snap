// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PinPage Snapshot Test renders correctly 1`] = `
<div
  className="pinPageBody"
>
  <div
    className="pinPageContainer"
  >
    <h1
      id="logo"
    >
      LOADING...
    </h1>
    <input
      className="pinPageInput"
      onChange={[Function]}
      onKeyDown={[Function]}
      pattern="\\d*"
      placeholder="PIN-kode ..."
      type="text"
      value=""
    />
    <button
      className="pinPageButton"
      onClick={[Function]}
    >
      Bli med i spillet!
    </button>
  </div>
</div>
`;
